TESTS/mbed_hal/lp_ticker/lp_ticker_api_tests.h

```diff
@@ -0,0 +1,56 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2017-2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/** \addtogroup hal_lp_ticker_tests
+ *  @{
+ */
+
+#ifndef LP_TICKER_API_TESTS_H
+#define LP_TICKER_API_TESTS_H
+
+#include "device.h"
+
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** Test that the ticker has the correct frequency and number of bits.
+ *
+ * Given ticker is available.
+ * When ticker information data is obtained.
+ * Then collected data indicates that ticker frequency is between 8KHz and 64KHz and the counter is at least 12 bits wide.
```

Comment:
Why setting a high limit to the ticker's frequency ?

Reply:
We should ask Russ since he is author of the ticker requirements. 
@c1728p9 @ithinuel

Reply:
This is the note in the header file `low power ticker: 8 KHz (1 tick per 125 us) - 64 KHz (1 tick per ~15.6 us)` 

My assumption is that with the upper limit, we have a guarantee that at least within this period of time, we have a tick (having tick match).